hack/metrics/metrics.go

```diff
@@ -0,0 +1,140 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"time"
+
+	_ "cloud.google.com/go/storage"
+	"contrib.go.opencensus.io/exporter/stackdriver"
+	"github.com/pkg/errors"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"go.opencensus.io/trace"
+	pkgtrace "k8s.io/minikube/pkg/trace"
+)
+
+const (
+	customMetricName = "custom.googleapis.com/minikube/start_time"
+	profile          = "cloud-monitoring"
+)
+
+var (
+	// The task latency in seconds
+	latencyS = stats.Float64("repl/start_time", "start time in seconds", "s")
+)
+
+func main() {
+	if err := execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func execute() error {
+	projectID := os.Getenv(pkgtrace.ProjectEnvVar)
+	if projectID == "" {
+		return fmt.Errorf("metrics collector requires a valid GCP project id set via the %s env variable", pkgtrace.ProjectEnvVar)
+	}
+
+	osMethod, err := tag.NewKey("os")
+	if err != nil {
+		return errors.Wrap(err, "new tag key")
+	}
+
+	// Register the view. It is imperative that this step exists,
+	// otherwise recorded metrics will be dropped and never exported.
+	v := &view.View{
+		Name:        customMetricName,
+		Measure:     latencyS,
+		Description: "minikube start time",
+		Aggregation: view.LastValue(),
+		TagKeys:     []tag.Key{osMethod},
+	}
+	if err := view.Register(v); err != nil {
+		return errors.Wrap(err, "registering view")
+	}
+
+	sd, err := stackdriver.NewExporter(stackdriver.Options{
+		ProjectID: projectID,
+		// MetricPrefix helps uniquely identify your metrics.
+		MetricPrefix: "minikube_start",
+		// ReportingInterval sets the frequency of reporting metrics
+		// to stackdriver backend.
+		ReportingInterval: 1 * time.Second,
+	})
+	if err != nil {
+		return errors.Wrap(err, "getting stackdriver exporter")
+	}
+	// It is imperative to invoke flush before your main function exits
+	defer sd.Flush()
+
+	// Register it as a trace exporter
+	trace.RegisterExporter(sd)
+
+	if err := sd.StartMetricsExporter(); err != nil {
+		return errors.Wrap(err, "starting metric exporter")
+	}
+	defer sd.StopMetricsExporter()
+	ctx := context.Background()
+	for {
+		st, err := minikubeStartTime(ctx, projectID)
+		if err != nil {
+			log.Printf("error collecting start time: %v", err)
+			continue
+		}
+		fmt.Printf("Latency: %f\n", st)
+		stats.Record(ctx, latencyS.M(st))
+		time.Sleep(30 * time.Second)
+	}
+}
+
+func minikubeStartTime(ctx context.Context, projectID string) (float64, error) {
+	minikubePath, err := downloadMinikube()
+	if err != nil {
+		return 0, errors.Wrap(err, "downloading minikube")
+	}
+	defer os.Remove(minikubePath)
+	defer deleteMinikube(ctx, minikubePath)
+
+	cmd := exec.CommandContext(ctx, minikubePath, "start", "--driver=docker", "-p", profile, "--memory=2000", "--trace=gcp")
```

Comment:
is there a reason to do 2000mb ram ? why not more realistic one, on most systems we auto-select way more than 2GB

Reply:
I want to measure the minimum required memory so that we can compare across platforms (linux/cloud shell/windows) with a constant variable.

I also don't want minikube to auto-select a high memory, which improves start time, and falsely lets us believe that minikube is fast when maybe it is slow with our minimum requirement of 2GB.

Reply:
sounds good, I am ok with this, however if the number is too slow, we could try a more realistic number like 4GB (as we choose in our functional test)